### PR TITLE
Use 'is' or 'is not' for comparisons to `None`

### PR DESCRIPTION
**Pull request by delirious-lettuce**
(cherry picked from commit 124f130)

> [Comparisons to singletons like None should always be done with is or is not, never the equality operators.](http://pep8.org/#programming-recommendations)

### CODE REVIEW

**Comment by TobyRoseman** [comment]
@delirious-lettuce - Thanks you for the pull request. When I try running the unit tests, with these changes, I get nine failures. I suspect this is due to the fact that `==` can be overridden by a magic method where `is` can not.

**Comment by delirious-lettuce** [comment]
@TobyRoseman ,

Hmm, I'll go and take a look and see if I can figure out which ones are causing this issue.

File: src/unity/python/turicreate/data_structures/sframe.py [hunk 1]
```diff
@@ -4393,7 +4393,7 @@ def filter_by(self, values, column_name, exclude=False):
                 tmp = SFrame(_proxy=self.__proxy__.join(value_sf.__proxy__,
                                                      'left',
                                                      {column_name:column_name}))
-                ret_sf = tmp[tmp[id_name] == None]
+                ret_sf = tmp[tmp[id_name] is None]
```
Comment (gustavla) [inline src/unity/python/turicreate/data_structures/sframe.py]: Try reverting this one and run the tests again.

File: src/unity/python/turicreate/test/test_knn_classifier.py [hunk 2]
```diff
@@ -486,8 +486,8 @@ def test_classify(self):
         ystar = self.model.classify(self.sf_test, max_neighbors=None,
                                     radius=1e-6, verbose=False)
 
-        self.assertTrue(all(ystar['class'] == None))
-        self.assertTrue(all(ystar['probability'] == None))
+        self.assertTrue(all(ystar['class'] is None))
+        self.assertTrue(all(ystar['probability'] is None))
```
Comment (gustavla) [inline src/unity/python/turicreate/test/test_knn_classifier.py]: Revert these as well.

File: src/unity/python/turicreate/test/test_knn_classifier.py [hunk 3]
```diff
@@ -585,8 +585,8 @@ def test_evaluate(self):
         cf_mat = evals['confusion_matrix']
 
         self.assertTrue(acc == 0.)
-        self.assertTrue(all(cf_mat['target_label'] != None))
-        self.assertTrue(all(cf_mat['predicted_label'] == None))
+        self.assertTrue(all(cf_mat['target_label'] is not None))
+        self.assertTrue(all(cf_mat['predicted_label'] is None))
```
Comment (gustavla) [inline src/unity/python/turicreate/test/test_knn_classifier.py]: And these two.

File: src/unity/python/turicreate/test/test_sarray.py [hunk 4]
```diff
@@ -843,8 +843,8 @@ def test_scalar_operators(self):
         self.__test_equal(s * 2, [array.array('d', [float(j) * 2 for j in i]) for i in self.vec_data], array.array)
         self.__test_equal(s / 2, [array.array('d', [float(j) / 2 for j in i]) for i in self.vec_data], array.array)
         s = SArray([1,2,3,4,None])
-        self.__test_equal(s == None, [0,0,0,0,1], int)
-        self.__test_equal(s != None, [1,1,1,1,0], int)
+        self.__test_equal(s is None, [0, 0, 0, 0, 1], int)
+        self.__test_equal(s is not None, [1, 1, 1, 1, 0], int)
```
Comment (gustavla) [inline src/unity/python/turicreate/test/test_sarray.py]: These two lines as well.

**Comment by gustavla** [comment]
Thanks for the commit @delirious-lettuce! I think @TobyRoseman is right and the problem is for cases where SFrame overloads the `==` operator. I will run the tests...

**Comment by delirious-lettuce** [comment]
@gustavla ,

No problem! Thanks for pointing them out!

Sorry for the delay, I've been setting up a new computer and was having trouble getting the project setup properly on Windows. Should have my Ubuntu virtual machine ready in a few minutes.

**Comment by gustavla** [comment]
Thanks @delirious-lettuce! I ran the full test suite and I'm finding a few more places that need to be reverted. I'll add comments to the lines as I find them.

**Comment by delirious-lettuce** [comment]
Sure, sounds good! I'll fix them ASAP.

**Comment by delirious-lettuce** [comment]
@gustavla ,

Thanks again for all your help!

**Pull request merged**